kaolin/csrc/cuda/ops/conversions/unbatched_mcube/unbatched_mcube_cuda_kernel.cu

```diff
@@ -491,28 +495,52 @@ generateTriangles2(float *pos, int *faces, int *compactedVoxelArray,
+    }
 }
 
 void launch_generateTriangles2(at::Tensor pos, at::Tensor faces, at::Tensor compactedVoxelArray,
-                              at::Tensor numTrianglesScanned, at::Tensor numPartialVertsScanned, at::Tensor numPartialVerts,
-                              at::Tensor voxelVertsOrder,
-                              int3 gridSize, at::Tensor voxelgrid,
-                              float3 voxelSize, float isoValue, int activeVoxels, int maxVerts)
+                                at::Tensor numTrianglesScanned, at::Tensor numPartialVertsScanned, at::Tensor numPartialVerts,
+                                at::Tensor voxelVertsOrder,
+                                int3 gridSize, at::Tensor voxelgrid,
+                                float3 voxelSize, float isoValue, int activeVoxels, int maxVerts)
 {
-  dim3 grid2((int) ceil(activeVoxels/ (float) NTHREADS), 1, 1);
+    dim3 grid2((int) ceil(activeVoxels/ (float) NTHREADS), 1, 1);
 
-  while (grid2.x > 65535) {
+    while (grid2.x > 65535) {
     grid2.x/=2;
     grid2.y*=2;
-  }
-
-  generateTriangles2<<<grid2, NTHREADS>>>(pos.data_ptr<float>(), faces.data_ptr<int>(),
-                                          compactedVoxelArray.data_ptr<int>(), numTrianglesScanned.data_ptr<int>(),
-                                          numPartialVertsScanned.data_ptr<int>(), numPartialVerts.data_ptr<int>(),
-                                          voxelVertsOrder.data_ptr<int>(),
-                                          voxelgrid.data_ptr<float>(), gridSize,
-                                          voxelSize, isoValue, activeVoxels,
-                                          maxVerts);
+    }
+
+    generateTriangles2<<<grid2, NTHREADS>>>(pos.data_ptr<float>(), faces.data_ptr<int>(),
+                                            compactedVoxelArray.data_ptr<int>(), numTrianglesScanned.data_ptr<int>(),
+                                            numPartialVertsScanned.data_ptr<int>(), numPartialVerts.data_ptr<int>(),
+                                            voxelVertsOrder.data_ptr<int>(),
+                                            voxelgrid.data_ptr<float>(), gridSize,
+                                            voxelSize, isoValue, activeVoxels,
+                                            maxVerts);
 }
+
+void CubScanWrapper(at::Tensor output, at::Tensor input, int numElements)
+{
+    int *d_in = input.data_ptr<int>();
+    int *d_out = output.data_ptr<int>();
+
+    cudaError_t err = cudaGetLastError();
+
+    if ( err != cudaSuccess )
+    {
+    printf("CUDA Error: %s\n", cudaGetErrorString(err));
+    exit(-1);
+    }
+    // Determine temporary device storage requirements
+    void *d_temp_storage = NULL;
+    size_t temp_storage_bytes = 0;
+    kaolin::cub::DeviceScan::ExclusiveSum(d_temp_storage, temp_storage_bytes, d_in, d_out, numElements);
+
+    // Allocate temporary storage
+    cudaMalloc(&d_temp_storage, temp_storage_bytes);
```

Comment:
Avoid cudaMalloc, prefer torch::empty / zeros.

Reply:
Not exactly sure how to use torch::empty/zeros here: https://nvlabs.github.io/cub/structcub_1_1_device_scan.html#a02b2d2e98f89f80813460f6a6ea1692b

Reply:
fixed